### PR TITLE
Change time-related fields type to decimal

### PR DESCRIPTION
**Pull request by kalaiyarasiganeshalingam**
## Purpose
$Subject

Couldn't change the type in method signature due to t[his](https://github.com/ballerina-platform/ballerina-lang/issues/28959).

Resolves [#967](https://github.com/ballerina-platform/ballerina-standard-library/issues/967)

### CODE REVIEW

File: cache-ballerina/abstract_cache.bal [hunk 1]
```diff
@@ -23,11 +23,11 @@ public type AbstractCache object {
     #
     # + key - Key of the value to be cached
     # + value - Value to be cached
-    # + maxAgeInSeconds - The time in seconds during which the cache entry is valid. '-1' means, the entry is valid
+    # + maxAge - The time in seconds during which the cache entry is valid. '-1' means, the entry is valid
     #                     forever
```
Comment (ldclakmal) [inline cache-ballerina/abstract_cache.bal:27]:
```suggestion
    # + maxAge - The time in seconds during which the cache entry is valid. '-1' means, the entry is valid forever
```

File: cache-ballerina/cache.bal [hunk 2]
```diff
@@ -23,16 +23,16 @@ import ballerina/time;
 # + capacity - Maximum number of entries allowed in the cache
 # + evictionFactor - The factor by which the entries will be evicted once the cache is full
 # + evictionPolicy - The policy which is used to evict entries once the cache is full
-# + defaultMaxAgeInSeconds - The default value in seconds which all the cache entries are valid.
+# + defaultMaxAge - The default value in seconds which all the cache entries are valid.
 #                            '-1' means, the entries are valid forever. This will be overwritten by the the
 #                            `maxAgeInSeconds` property set when inserting item to the cache
```
Comment (ldclakmal) [inline cache-ballerina/cache.bal:28]: Shall we format this documentation?

File: cache-ballerina/cache.bal [hunk 3]
```diff
@@ -23,16 +23,16 @@ import ballerina/time;
...
-# + cleanupIntervalInSeconds - Interval of the timer task, which will clean up the cache
+# + cleanupInterval - Interval of the timer task, which will clean up the cache
```
Comment (ldclakmal) [inline cache-ballerina/cache.bal:29]:
```suggestion
# + cleanupInterval - Interval (in seconds) of the timer task, which will clean up the cache
```

File: cache-ballerina/cache.bal [hunk 4]
```diff
@@ -93,18 +93,18 @@ public class Cache {
...
 
-        int? cleanupIntervalInSeconds = cacheConfig?.cleanupIntervalInSeconds;
-        if (cleanupIntervalInSeconds is int) {
+        decimal? interval = cacheConfig?.cleanupInterval;
+        if (interval is decimal) {
             task:TimerConfiguration timerConfiguration = {
-                intervalInMillis: cleanupIntervalInSeconds,
-                initialDelayInMillis: cleanupIntervalInSeconds
+                intervalInMillis: <int> interval,
+                initialDelayInMillis: <int> interval
```
Comment (ldclakmal) [inline cache-ballerina/cache.bal:107]: If we change the types defined in `task` module first, we may not need to cast the values to `int`. WDYT?

Comment (kalaiyarasiganeshalingam) [reply cache-ballerina/cache.bal:107]: We are working on revamping the task module. Once that finish, we need to update all task-related changes.  Therefore, We skipped this.

**Review by ldclakmal — APPROVED** [review]

**Pull request merged**